V2 - Initial tab activation

ouchadam opened the pull request:
Fixes crash when not setting/updating the landingstrip adapter before setting a viewpager adapter after setting the `landingstrip` as a `onPageChangedListener` -

`viewpager.setAdapter` triggers `onPageScrolled` however we may not have set or updated a `TabAdapter` with content causing the `ScrollingPageChangeListener` to crash as the `TabContainerView` contains no tabs! 

fixed by guarding the `OnPageChangedListener` methods with `tabsContainerView.getChildCount() == 0`

Fixes initial tab activated state being skipped -

By ignoring scroll events when the tabs are missing, we broke the initial tab activation state from being set due to a hidden race condition in `onPageScrolled -> TabsContainerView.setActivated`

This is fixed by moving the initial tab activation state to after notifying/creating the tabs and by introducing a new state `selectedPosition` which is used to restore the position state

Commit:
ignoring scroll events when the tabs contains no children

Commit:
moving the initial post adapter set activated position setting

Commit:
adding the initial page activated setting for when the adapter has been set/updated for the first time

Commit:
adding the initial page activated setting for when the adapter has been set/updated for the first time

Commit:
only applying the initial adapter activated state when notifying the entire dataset as we only need to do it when creating the tabs

ataulm commented on core/src/main/java/com/novoda/landingstrip/Notifier.java:
```diff
@@ -25,6 +28,13 @@ private void recreateAndBindTabs(BaseAdapter<T> adapter) {
         }
     }
 
+    private void handleAdapterSetBecausePageSelectedIsNotCalled(BaseAdapter<T> adapter) {
```
should change the method name here - the ScrollingPageChangeListener knows about PageSelected since it implements OnPageChangeListener. This class doesn't.

How about simply `updateActivatedTab()` or `syncActivatedTab`?

ataulm commented on core/src/main/java/com/novoda/landingstrip/Notifier.java:
```diff
@@ -25,6 +28,13 @@ private void recreateAndBindTabs(BaseAdapter<T> adapter) {
     }
 
+    private void handleAdapterSetBecausePageSelectedIsNotCalled(BaseAdapter<T> adapter) {
+        if (state.firstTimeAccessed() && adapter.getCount() > 0) {
```
the `adapter.getCount() > 0` check could fit in `onNotifyDataSetChanged()`, then this method would be simpler.

Commit:
setting demo tab text color with normal and activated states

Commit:
adding the selected postion to the state as this comes from two sources -
on scroll and on page selected, when resetting the adapter we should only use the selected position

Commit:
simplifying the tabscontainerview init as it activates tabs based on the child count

Commit:
removing the unused firsttimeaccessed state, it has been replaced with selectedPosition

Electryc reviewed (approved):
Looking good! 👍

ataulm commented on core/src/main/java/com/novoda/landingstrip/State.java:
```diff
@@ -8,5 +8,5 @@
 
     int fastForwardPosition();
 
-    boolean firstTimeAccessed();
```
:+1:
